Instagram/Instagram/Source/Screen/Auth/Controller/UserNameViewController.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  SignUpViewController.swift
+//  Instagram
+//
+//  Created by 소연 on 2022/04/04.
+//
+
+import UIKit
+
+import RxSwift
+
+import SnapKit
+import Then
+
+final class UserNameViewController: UIViewController {
+    
+    // MARK: - Properties
+
+    var disposeBag = DisposeBag()
+    
+    private lazy var navigationBar = IDSNavigationBar(self,
+                                                      view: .auth,
+                                                      backButtonIsHidden: false,
+                                                      closeButtonIsHidden: true)
+    
+    private lazy var rootView = SignUpView()
+    
+    // MARK: - Life Cycle
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.isNavigationBarHidden = true
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configUI()
+        setLayout()
+        bind()
+    }
+    
+    // MARK: - Init UI
+    
+    private func configUI() {
+        view.backgroundColor = .white
+    }
+    
+    private func setLayout() {
+        view.addSubviews([navigationBar, rootView])
+        
+        navigationBar.snp.makeConstraints {
+            $0.top.left.right.equalTo(view.safeAreaLayoutGuide)
+            $0.height.equalTo(50)
+        }
```

Comment:
또 하나의 FYI
이것도 자라나라머리머리 하면서 알게된 사실인데, 
SnapKit을 만든 사람이 코드를 직관적으로 쓰게 하기 위해서 영어 문장 형식으로 읽히게끔 만들어놨다고 해.

그래서 보통 make를 써줘서 문장 형식을 맞춰주는 편이야
읽어보면 Make centerX equal to superview (centerX를 superview와 같게 해라) 이렇게 읽힐 수 있게 만든 라이브러리라고해!

```suggestion
        navigationBar.snp.makeConstraints { make in 
            make.top.left.right.equalTo(view.safeAreaLayoutGuide)
            make.height.equalTo(50)
        }
```

Reply:
어머나 그렇군 .. 몰랐던 사실이네요 .. 이것도 습관을 버리지 못한 .. 참고하겠습니다 !!